examples/allinone/05_up.sh

```diff
@@ -1,4 +1,26 @@
 #!/bin/bash
 # This is the script for bringing up the standard openstack nodes without
 # Swift. This is probably the up script you want to run.
-vagrant up --provider vmware_fusion
+
+PROVIDER=${PROVIDER:-vmware_fusion}
+
+#---  FUNCTION  ----------------------------------------------------------------
+#          NAME:  provider
+#   DESCRIPTION:  checks for binary name in PATH
+#    PARAMETERS:  -
+#       RETURNS:  identifier for provider
+#-------------------------------------------------------------------------------
+provider ()
+{
+    if [[ -x `which vmware_fusion` ]]
+    then
+        echo "vmware_fusion"
+    elif [[ -x `which VirtualBox` ]]
+    then
+        echo "virtualbox"
+    else
+        exit 1
+    fi
+}   # ----------  end of function provider  ----------
+
```

Comment:
The provider() function doesn't use the $PROVIDER variable that you set above. Shouldn't it try to use $PROVIDER first if it is set by the user?